File: azure-blobstore/AzureBlobService.cs
using System.Collections.Generic;
using System.Diagnostics;
using Microsoft.WindowsAzure.Storage;
using Microsoft.WindowsAzure.Storage.Blob;
using Microsoft.Extensions.Configuration;
using System;
using System.IO;

namespace com.opusmagus.cloud.blobs {
    public class AzureBlobService : IBlobService {

        public AzureBlobService() {

        }
        public AzureBlobService(IConfiguration config) {

        }

        public IEnumerable<IListBlobItem> getBlobItems(string blobContainerConnString, string blobContainerName, int maxResults)
        {
            var storageAccount = CloudStorageAccount.Parse(blobContainerConnString);
            var blobClient = storageAccount.CreateCloudBlobClient();
            var container = blobClient.GetContainerReference(blobContainerName);
            var blobItems = container.ListBlobsSegmentedAsync("", true, BlobListingDetails.All, maxResults, null, null, null).ConfigureAwait(false).GetAwaiter().GetResult().Results;            
            return blobItems;
        }        

        public byte[] getBlobContents(string blobContainerConnString, string blobContainerName, string blobName) {
            var storageAccount = CloudStorageAccount.Parse(blobContainerConnString);
            var blobClient = storageAccount.CreateCloudBlobClient();
            var container = blobClient.GetContainerReference(blobContainerName);
            var blobBlockRef = container.GetBlockBlobReference(blobName);
            blobBlockRef.FetchAttributesAsync().ConfigureAwait(false).GetAwaiter().GetResult();
            byte[] buffer = new byte[blobBlockRef.Properties.Length];       
            blobBlockRef.DownloadToByteArrayAsync(buffer, 0).ConfigureAwait(false).GetAwaiter().GetResult();
            return buffer;
        }

        private byte[] getBlobContentsOld(string blobContainerConnString, string blobContainerName)
        {
            //var urls = new List<string>();            
            //string blobContainerName = "sample-blob-container";
            var storageAccount = CloudStorageAccount.Parse(blobContainerConnString);
            var blobClient = storageAccount.CreateCloudBlobClient();
            var container = blobClient.GetContainerReference(blobContainerName);
            var blobItems = container.ListBlobsSegmentedAsync("", true, BlobListingDetails.All, 200, null, null, null).ConfigureAwait(false).GetAwaiter().GetResult().Results;
            foreach(var blobItem in blobItems) {
                Debug.WriteLine($"blobUri:{blobItem.Uri}");
                var blockBlob = (CloudBlockBlob)blobItem;
                byte[] buffer = new byte[blockBlob.Properties.Length];                
                var blobRef = container.GetBlobReference(blobItem.Uri.ToString());
                //blockBlob.DownloadToFileAsync($"./local/{Guid.NewGuid()}.docx", FileMode.CreateNew).ConfigureAwait(false).GetAwaiter().GetResult();
                blockBlob.DownloadToByteArrayAsync(buffer, 0).ConfigureAwait(false).GetAwaiter().GetResult();
                return buffer;
            }
            
            return null;
        }
    }
}

File: java/az-blob-storage-layer/src/main/java/dk/commentor/dal/blobstore/AzureBlobService.java
package dk.commentor.dal.blobstore;

import java.nio.charset.StandardCharsets;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionException;

import com.microsoft.azure.storage.CloudStorageAccount;
import com.microsoft.azure.storage.ResultSegment;
import com.microsoft.azure.storage.blob.CloudBlobContainer;
import com.microsoft.azure.storage.blob.CloudBlockBlob;
import com.microsoft.azure.storage.blob.ListBlobItem;
import com.microsoft.azure.storage.blob.CloudBlobClient;

public class AzureBlobService {
    private final CloudStorageAccount storageAccount;
    private final CloudBlobClient blobClient;

    public AzureBlobService(String blobContainerConnString) throws Exception {
        storageAccount = CloudStorageAccount.parse(blobContainerConnString);
        blobClient = storageAccount.createCloudBlobClient();
    }

    public CompletableFuture<ResultSegment<ListBlobItem>> getBlobItems(String blobContainerName, Integer maxResults) throws Exception {
        // BlobListingDetails.SNAPSHOTS        
        return CompletableFuture.supplyAsync(() -> {
            try {
                CloudBlobContainer container = blobClient.getContainerReference(blobContainerName);
                ResultSegment<ListBlobItem> blobItems = container.listBlobsSegmented("", true, null, maxResults, null, null,
                        null);
                return blobItems;
            }
            catch (Exception ex) {
                throw new CompletionException(ex);
            }
        });        
    }

    public CompletableFuture<byte[]> getBlobContents(String blobContainerName, String blobName) throws Exception {
        return CompletableFuture.supplyAsync(() -> {
            CloudBlobContainer container;
            try {
                container = blobClient.getContainerReference(blobContainerName);
                CloudBlockBlob blobBlockRef = container.getBlockBlobReference(blobName);
                blobBlockRef.downloadAttributes();
                byte[] buffer = new byte[(int)blobBlockRef.getProperties().getLength()];       
                blobBlockRef.downloadToByteArray(buffer, 0);
                return buffer;
            } 
            catch (Exception ex) {
                throw new CompletionException(ex);
            }
        });
    }

    public CompletableFuture<String> getBlobContentsAsString(String blobContainerName, String blobName) throws Exception
    {
        return CompletableFuture.supplyAsync( () -> {
            try {
                byte[] buffer = getBlobContents(blobContainerName, blobName).get();
                String stringBuffer  = new String(buffer, StandardCharsets.UTF_8);
                return stringBuffer;
            } 
            catch (Exception ex) {
                throw new CompletionException(ex);
            }
        });
    }
        
    public CompletableFuture<Void> putBlobContents(String blobContainerName, String blobName, String jsonData) throws Exception
    {
        return CompletableFuture.runAsync( () -> {
            try {
                CloudBlobContainer container = blobClient.getContainerReference(blobContainerName);
                CloudBlockBlob blobBlockRef = container.getBlockBlobReference(blobName);
                blobBlockRef.downloadAttributes();
                blobBlockRef.uploadText(jsonData);
            } 
            catch (Exception ex) {
                throw new CompletionException(ex);
            }
        });
    }
}
